### ISSUE TITLE
npm run sites should check if the images exist and run the build:images

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#1997]
Currently we expect people to know to run `npm run build:images` separately before running `npm run site` the first time, which is non-trivial.

**Comment by YuhanLu** [comment]
I can think of 2 ways to do this:
1. automatically run `npm run build:images` when people don't have all images under `build/examples/images`
2. Always have a warning message to alert people to run `npm run build:images` by themselves.

Which one should we follow?

**Comment by domoritz** [comment]
I don't think this is important enough. Our build system requires non-trivial steps regardless of what we try to do with just package.json. Especially this particular issue is not that bad because it's obvious that images are not being generated and it doesn't have any effects if somebody submits a PR without running this command.

**Comment by kanitw** [comment]
Adding a warning message should take 10 mins, so lets do that and add note to contributor.md.

Guessing why its broken would be annoying.

**Comment by kanitw** [comment]
Actually autoamtically run build images is better and should take 10 mins to do too

**Comment by kanitw** [comment]
This is definitely an annoying problem  This is like my tenth time getting confused because of this problem.  It will definitely many other contributors.

**Comment by domoritz** [comment]
Sounds like we need a makefile.
